File: lib/src/main/java/com/sd/lib/stream/factory/SimpleDefaultStreamFactory.java
package com.sd.lib.stream.factory;

import androidx.annotation.NonNull;

import com.sd.lib.stream.FStream;

public class SimpleDefaultStreamFactory implements DefaultStreamFactory {
    @NonNull
    @Override
    public FStream create(@NonNull CreateParam param) {
        try {
            return param.classDefaultStream.newInstance();
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}

File: lib/src/main/java/com/sd/lib/stream/factory/SimpleDefaultStreamFactory.kt
package com.sd.lib.stream.factory

import com.sd.lib.stream.FStream
import com.sd.lib.stream.factory.DefaultStreamFactory.CreateParam

open class SimpleDefaultStreamFactory : DefaultStreamFactory {

    override fun create(param: CreateParam): FStream {
        try {
            return param.classStreamDefault.newInstance()
        } catch (e: Exception) {
            throw RuntimeException(e)
        }
    }
}